<launch>
  <arg name="cpu_monitor_config_file" default="$(find-pkg-share autoware_system_monitor)/config/cpu_monitor.param.yaml"/>
  <arg name="hdd_monitor_config_file" default="$(find-pkg-share autoware_system_monitor)/config/hdd_monitor.param.yaml"/>
  <arg name="mem_monitor_config_file" default="$(find-pkg-share autoware_system_monitor)/config/mem_monitor.param.yaml"/>
  <arg name="net_monitor_config_file" default="$(find-pkg-share autoware_system_monitor)/config/net_monitor.param.yaml"/>
  <arg name="ntp_monitor_config_file" default="$(find-pkg-share autoware_system_monitor)/config/ntp_monitor.param.yaml"/>
  <arg name="process_monitor_config_file" default="$(find-pkg-share autoware_system_monitor)/config/process_monitor.param.yaml"/>
  <arg name="gpu_monitor_config_file" default="$(find-pkg-share autoware_system_monitor)/config/gpu_monitor.param.yaml"/>
  <arg name="voltage_monitor_config_file" default="$(find-pkg-share autoware_system_monitor)/config/voltage_monitor.param.yaml"/>

  <group>
    <node_container pkg="rclcpp_components" exec="component_container_mt" name="system_monitor_container" namespace="system_monitor" output="screen">
      <composable_node pkg="autoware_system_monitor" plugin="CPUMonitor" name="cpu_monitor">
        <param from="$(var cpu_monitor_config_file)"/>
      </composable_node>
      <composable_node pkg="autoware_system_monitor" plugin="MemMonitor" name="mem_monitor">
        <param from="$(var mem_monitor_config_file)"/>
      </composable_node>
      <composable_node pkg="autoware_system_monitor" plugin="NetMonitor" name="net_monitor">
        <param from="$(var net_monitor_config_file)"/>
      </composable_node>
      <composable_node pkg="autoware_system_monitor" plugin="NTPMonitor" name="ntp_monitor">
        <param from="$(var ntp_monitor_config_file)"/>
      </composable_node>
      <composable_node pkg="autoware_system_monitor" plugin="ProcessMonitor" name="process_monitor">
        <param from="$(var process_monitor_config_file)"/>
      </composable_node>
      <composable_node pkg="autoware_system_monitor" plugin="HddMonitor" name="hdd_monitor">
        <param from="$(var hdd_monitor_config_file)"/>
      </composable_node>
      <composable_node pkg="autoware_system_monitor" plugin="GPUMonitor" name="gpu_monitor">
        <param from="$(var gpu_monitor_config_file)"/>
      </composable_node>
      <composable_node pkg="autoware_system_monitor" plugin="VoltageMonitor" name="voltage_monitor">
        <param from="$(var voltage_monitor_config_file)"/>
      </composable_node>
    </node_container>
  </group>
</launch>
